cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -40,23 +43,55 @@ case class CommandLineTool private(
                                    temporaryFailCodes: Option[Array[Int]],
                                    permanentFailCodes: Option[Array[Int]]) {
 
-  def womExecutable(inputFile: Option[String] = None): Checked[Executable] =
-    CwlExecutableValidation.buildWomExecutable(ExecutableTaskDefinition.tryApply(taskDefinition).toEither, inputFile)
+  /** Builds an `Executable` directly from a `CommandLineTool` CWL with no parent workflow. */
+  def womExecutable(validator: Requirement => ErrorOr[Requirement], inputFile: Option[String] = None): Checked[Executable] = {
+    val taskDefinition = buildTaskDefinition(parentWorkflow = None, validator)
+    CwlExecutableValidation.buildWomExecutable(taskDefinition, inputFile)
+  }
+
+  private def validateRequirementsAndHints(parentWorkflow: Option[Workflow], validator: Requirement => ErrorOr[Requirement]): ErrorOr[List[Requirement]] = {
+    import cats.instances.list._
+    import cats.syntax.traverse._
+
+    val allRequirements = requirements.toList.flatten ++ parentWorkflow.toList.flatMap(_.allRequirements)
+    // All requirements must validate or this fails.
+    val errorOrValidatedRequirements: ErrorOr[List[Requirement]] = allRequirements traverse validator
```

Comment:
So this is perhaps a bit aggressive as it will validate requirements at the workflow level even if they are always overridden by requirements specified in subworkflows or tools.